quora-service/src/main/java/com/upgrad/quora/service/business/AnswerBusinessService.java

```diff
@@ -48,4 +50,41 @@ public AnswerEntity createAnswer(final AnswerEntity answerEntity, final String q
         answerEntity.setQuestion(questionEntity);
         return answerDao.createAnswer(answerEntity);
     }
+
+    public void deleteAnswer(final String uuid, final String authorization) throws AuthorizationFailedException, AnswerNotFoundException {
+
+        UserAuthEntity userAuthEntity = userDao.getUserAuthToken(authorization);
+        if (userAuthEntity == null) {
+            throw new AuthorizationFailedException("ATHR-001", "'User has not signed in");
+
+        }
+        if (userAuthEntity.getLogoutAt() != null) {
+            throw new AuthorizationFailedException("ATHR-002", "User is signed out");
+        }
+        if(userAuthEntity.getUser().getRole().equals("nonadmin") || !userAuthEntity.getUser().equals(answerDao.getAnswerByUuid(uuid).getUser())){
+            throw new AuthorizationFailedException("ATHR-003", "Only the answer owner or admin can delete the answer");
+
+        }
+        if (answerDao.getAnswerByUuid(uuid)==null) {
+            throw new AnswerNotFoundException("AND-001","Entered answer uuid does not exist.");
+        }
+
+        answerDao.userQuestionDelete(uuid);
+    }
+
```

Comment:
Add this line above getAllAnswersToQuestion method:

```suggestion
@Transactional(propagation = Propagation.REQUIRED)
```